fastparquet/test/test_read.py

```diff
@@ -431,3 +431,25 @@ def test_truncated_decimal():
         name='weight measure:WEIGHT(KG, 0)')
     out = df['weight measure:WEIGHT(KG, 0)']
     assert np.allclose(expected, out)
+
+
+def test_or_filtering(tempdir):
```

Comment:
Any chance we can rip a set of filter conditions from pyarrow and ensure that they evaluate the same for us?

Reply:
Hi Martin, thanks for the quick review.
I am not sure to understand the word 'rip' in this context. Please, could you clarify?
Should I import pyarrow and make a read with `read_table` with some filters and then compare resulting pandas DataFrame? (it seems to me that it will call different functions that makes the resulting 'unit test' not so 'unit' anylonger but I can try)

One thing, [documentation of pyarrow filtering](http://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html) states:
'Rows which do not match the filter predicate will be removed from scanned data.'

So this is not 'row group filtering' but 'row filtering'.
In the end, to compare results, I will also have to truncate DataFrame from fastparquet, and in the end... not so sure if the comparison is so relevant.

Please, can you detail a bit more what addition you would like me to do?
Thanks in advance for your feedback.
Bests,

Reply:
I agree that it can be a bit of a pain to compare the row-wise filtering in pyarrow with the row-group filtering here.  One option (for the pyarrow.dataset API) is to define your own row-group-filtered pyarrow read for comparison.  For example:

```python
import pandas as pd
import numpy as np
import pyarrow as pa
from pyarrow import dataset as pa_ds

def pyarrow_rg_filtered_read(path, filters):
    ds = pa_ds.dataset(path)
    ds_filters = pa.parquet._filters_to_expression(filters)
    tables = []
    for file_frag in ds.get_fragments(ds_filters):
        for rg_frag in file_frag.split_by_row_group(ds_filters):
            tables.append(rg_frag.to_table())
    return pa.concat_tables(tables).to_pandas()

# Test case
# (Write 4 row-groups, and expect first and last to be read in)
path = "test.pq"
filters = [[("a", ">", 3),("a", "<", 5)],[("a", "==", 19)]]
pd.DataFrame({"a": np.arange(20)}).to_parquet(path, engine="pyarrow", row_group_size=5)
pyarrow_rg_filtered_read(path, filters)
```

**Result**
```
    a
0   0
1   1
2   2
3   3
4   4
5  15
6  16
7  17
8  18
9  19
```

Reply:
Yup - Agree that there is not much reason to use pyarrow in CI.